### PR TITLE
Toggleable exploiting log, Jail glitch log entry and disabled jumping

### PR DESCRIPTION
**Pull request by SimZor**
#### Changes proposed in this pull request: 
- Toggleable - Exploiting Log (Only for Jail in this pull request)
- Toggleable - Jail Glitch Teleport (If toggleable, will teleport back to the jail after glitching out)
- Toggleable - Disable Jumping

### CODE REVIEW

File: Altis_Life.Altis/core/functions/fn_keyHandler.sqf [hunk 1]
```diff
@@ -55,14 +55,19 @@ if (life_container_active) then {
...
+        if ((LIFE_SETTINGS(getNumber,"jail_disable_jumping") isEqualTo 1) && (life_is_arrested = true)) then {
+            hint "You cannot jump since you are wearing an anti-jump suit.";
+        } else {
+            if (isNil "jumpActionTime") then {jumpActionTime = 0;};
+            if (_shift && {!(animationState player isEqualTo "AovrPercMrunSrasWrflDf")} && {isTouchingGround player} && {stance player isEqualTo "STAND"} && {speed player > 2} && {!life_is_arrested} && {((velocity player) select 2) < 2.5} && {time - jumpActionTime > 1.5}) then {
+                jumpActionTime = time; //Update the time.
+                [player] remoteExec ["life_fnc_jumpFnc",RANY]; //Global execution
+                _handled = true;
+            };
+        }:
```
Comment (Jawshy) [inline Altis_Life.Altis/core/functions/fn_keyHandler.sqf]: `};`*

Comment (SimZor) [inline Altis_Life.Altis/core/functions/fn_keyHandler.sqf]: Fixed :)

File: Altis_Life.Altis/stringtable.xml [hunk 2]
```diff
@@ -901,6 +901,19 @@
             <German>Admin %1 hat die Debug-Konsole geöffnet.</German>
         </Key>
     </Package>
+    <Package name="Jail_Strings">
+        <Key ID="STR_JAIL_NOTIFYGLITCH">
+            <Original>You've magically glitched out of jail. This has been logged and more attempts may result in punishment.</Original>
+            <Czech>Vy jste magicky Odvázaný z vězení. Tato skutečnost byla zaznamenána a další pokusy může mít za následek trest.</Czech>
+            <Spanish>Usted ha glitched mágicamente fuera de la cárcel. Esto ha sido registrada y más intentos puede resultar en un castigo.</Spanish>
+            <Russian>Вы волшебным образом сглаженным из тюрьмы. Это было зарегистрировано и больше попыток может привести к наказанию.</Russian>
+            <German>Sie haben auf magische Weise aus dem Gefängnis glitched. Dies wurde protokolliert und weitere Versuche Bestrafung zur Folge haben.</German>
```
Comment (setoy) [inline Altis_Life.Altis/stringtable.xml]: `<German>Du hast dich magisch aus dem Gefängnis befreit. Dies wurde protokolliert und weitere Versuche könnten Bestrafungen zur Folge haben.</German>`

File: Altis_Life.Altis/core/functions/fn_keyHandler.sqf [hunk 3]
```diff
@@ -55,13 +55,18 @@ if (life_container_active) then {
...
+
     //Space key for Jumping
     case 57: {
-        if (isNil "jumpActionTime") then {jumpActionTime = 0;};
-        if (_shift && {!(animationState player isEqualTo "AovrPercMrunSrasWrflDf")} && {isTouchingGround player} && {stance player isEqualTo "STAND"} && {speed player > 2} && {!life_is_arrested} && {((velocity player) select 2) < 2.5} && {time - jumpActionTime > 1.5}) then {
-            jumpActionTime = time; //Update the time.
-            [player] remoteExec ["life_fnc_jumpFnc",RANY]; //Global execution
-            _handled = true;
+        if ((LIFE_SETTINGS(getNumber,"jail_disable_jumping") isEqualTo 1) && (life_is_arrested = true)) then {
+            hint "You cannot jump since you are wearing an anti-jump suit.";
```
Comment (Jawshy) [inline Altis_Life.Altis/core/functions/fn_keyHandler.sqf]: Please [localise](https://community.bistudio.com/wiki/localize) this `hint`.

Comment (SimZor) [inline Altis_Life.Altis/core/functions/fn_keyHandler.sqf]: Done.

File: Altis_Life.Altis/stringtable.xml [hunk 4]
```diff
@@ -901,6 +901,30 @@
...
+            <Italian>Hai magicamente glitched fuori di prigione. Questo è stato registrato e più tentativi può causare una punizione.</Italian>
+            <Portuguese>Você magicamente glitched fora da cadeia. Este foi registado e mais tentativas pode resultar em punição.</Portuguese>
+            <Polish>Masz magicznie glitched z więzienia. Zostało rejestrowane i więcej prób może skutkować karą.</Polish>
+        </Key>
+        <Key ID="STR_JAIL_NOTIFYANTIJUMP">
+            <Original>You cannot jump since you are wearing an anti-jump suit.</Original>
+            <Czech>Nemůžete skákat, protože jste na sobě anti-skok oblek.</Czech>
+            <Spanish>No se puede saltar desde que usted está usando un traje anti-salto.</Spanish>
+            <Russian>Вы не можете прыгать, так как вы носите анти-полукомбинезона.</Russian>
+            <German>Sie können nicht springen, da Sie ein Anti-Sprung-Anzug tragen.</German>
```
Comment (setoy) [inline Altis_Life.Altis/stringtable.xml]: Your wish is my command.

`<German>Du kannst nicht springen, da du einen Anti-Sprung-Anzug trägst.</German>`

**Comment by SimZor** [comment]
When is this being merged? :)

File: life_server/init.sqf [hunk 5]
```diff
@@ -151,6 +151,7 @@ TON_fnc_requestClientID =
 /* Event handler for logs */
 "money_log" addPublicVariableEventHandler {diag_log (_this select 1)};
 "advanced_log" addPublicVariableEventHandler {diag_log (_this select 1)};
+"exploit_log" addPublicVariableEventHandler {diag_log (_this select 1)};
```
Comment (SimZor) [inline life_server/init.sqf]: Just thought this could be used for future trapping of possible exploiters?

**Comment by SimZor** [comment]
@BoGuu is this okay now or did you want something else?

**Comment by Jawshy** [comment]
Booger always wants more. 

@SimZor, in the stringtable could you copy what you have for STR_DL_EL_JAILGLITCH_BEF and STR_DL_EL_JAILGLITCH to the other languages? Even if it is not translated at least the fields are not empty.

**Comment by SimZor** [comment]
Sure! @Jawsh

File: life_server/init.sqf (hunk 5 above)
Comment (BoGuu) [inline life_server/init.sqf]: Well, I mean, look at it...

All it does is diag_log what is passed, same as the others

File: Altis_Life.Altis/config/Config_Master.hpp [hunk 6]
```diff
@@ -95,6 +96,8 @@ class Life_Settings {
     jail_seize_vItems[] = { "spikeStrip","lockpick","goldbar","blastingcharge","boltcutter","defusekit","heroin_unprocessed","heroin_processed","cannabis","marijuana","cocaine_unprocessed","cocaine_processed","turtle_raw" }; //Define VIRTUAL items you want to be removed from players upon jailing here. Use "jail_seize_inventory" for Arma inventory items.
     jail_seize_inventory = false; //Set to true to run the cop seize script on inmates. False will remove only weapons and magazines otherwise. (Basically used in case cops forget to seize items). [See Lines 106-111 below]
     sendtoJail_locations[] = { "police_hq_1", "police_hq_2", "cop_spawn_3", "cop_spawn_5", "Correctional_Facility" }; //Enter the variableName from the mission.sqm here to allow cops to send a person to jail at these locations.
+    jail_forceWalk = true; // Disable running/jumping in jail to avoid glitchers
+    jail_helicopter_only = true; // If enabled (true) this will teleport players back into the jail if they're not in a helicopter to avoid jailglitches
```
Comment (BoGuu) [inline Altis_Life.Altis/config/Config_Master.hpp]: Didn't I say no to this one?

File: Altis_Life.Altis/core/civilian/fn_jailMe.sqf [hunk 7]
```diff
@@ -36,6 +36,27 @@ for "_i" from 0 to 1 step 0 do {
         hintSilent parseText format [(localize "STR_Jail_Time")+ "<br/> <t size='2'><t color='#FF0000'>%1</t></t><br/><br/>" +(localize "STR_Jail_Pay")+ " %3<br/>" +(localize "STR_Jail_Price")+ " $%2",_countDown,[life_bail_amount] call life_fnc_numberText,if (life_canpay_bail) then {"Yes"} else {"No"}];
     };
 
+    if (LIFE_SETTINGS(getNumber,"forceWalk") isEqualTo 1) then {
+        player forceWalk true;
```
Comment (BoGuu) [inline Altis_Life.Altis/core/civilian/fn_jailMe.sqf]: How do they ever run again!?

Comment (SimZor) [inline Altis_Life.Altis/core/civilian/fn_jailMe.sqf]: Fixed. Don't know how I could've forgotten that :D

File: Altis_Life.Altis/config/Config_Master.hpp (hunk 6 above)
Comment (SimZor) [inline Altis_Life.Altis/config/Config_Master.hpp]: Do you want me only to remove the toggle or to remove the helicopter teleport back?

File: Altis_Life.Altis/config/Config_Master.hpp [hunk 8]
```diff
@@ -95,8 +95,7 @@ class Life_Settings {
     /* Jail System Configurations */
     jail_seize_vItems[] = { "spikeStrip","lockpick","goldbar","blastingcharge","boltcutter","defusekit","heroin_unprocessed","heroin_processed","cannabis","marijuana","cocaine_unprocessed","cocaine_processed","turtle_raw" }; //Define VIRTUAL items you want to be removed from players upon jailing here. Use "jail_seize_inventory" for Arma inventory items.
     jail_seize_inventory = false; //Set to true to run the cop seize script on inmates. False will remove only weapons and magazines otherwise. (Basically used in case cops forget to seize items). [See Lines 106-111 below]
-    sendtoJail_locations[] = { "police_hq_1", "police_hq_2", "cop_spawn_3", "cop_spawn_5", "Correctional_Facility" }; //Enter the variableName from the mission.sqm here to allow cops to send a person to jail at these locations.
-    jail_forceWalk = true; // Disable running/jumping in jail to avoid glitchers
```
Comment (BoGuu) [inline Altis_Life.Altis/config/Config_Master.hpp]: Let's get this back, sorry for the confusion ;)

**Comment by SimZor** [comment]
My git is f-ed up atm

**Comment by SimZor** [comment]
f-ed up

**Comment by Jawshy** [comment]
See https://github.com/AsYetUntitled/Framework/pull/132.

**Pull request closed without merging**